float32[] joint_angles
float32[] base_pose